taichi/runtime/llvm/llvm_offline_cache.cpp

```diff
@@ -266,7 +266,7 @@ void LlvmOfflineCacheFileWriter::dump(const std::string &path,
       TI_ASSERT(mod != nullptr);
       if (format & Format::LL) {
         std::string filename = filename_prefix + ".ll";
-        if (try_lock_with_file(filename)) {  // Not exists
+        if (!merge_with_old || try_lock_with_file(filename)) {
```

Comment:
OOC: I took a quick look at how merging works, I wonder is there any specific reason that we need to explicitly load old metadata and overwrite instead of just overwriting without looking at the old ones? Thanks!

Reply:
Merging with old metadata is required by the offline cache.

For Offline Cache:
* Merging the new metadata with the old metadata is needed, otherwise we will lose old cached kernels.

see 
https://github.com/taichi-dev/taichi/blob/master/taichi/runtime/program_impls/llvm/llvm_program.cpp#L181-L183

For AOT: not needed and causes the bug.